documentation/modules/exploit/windows/http/tomcat_cgi_cmdlineargs.md

```diff
@@ -0,0 +1,102 @@
+## Description
+
+This module exploits a vulnerability in Apache Tomcat's CGIServlet component. When the enableCmdLineArguments setting is set to true, a remote user can abuse this to execute system commands, and gain remote code execution.
+
+## Vulnerable Application
+
+The following versions of Apache Tomcat on Windows are effected:
+
+* 9.0.0.M1 to 9.0.17
+* 8.5.0 to 8.5.39
+* 7.0.0 to 7.0.93
+
+Also, the machine needs to enable the enableCmdLineArguments option in web.xml. For example:
+
+```xml
+<servlet>
+<servlet-name>cgi</servlet-name>
+<servlet-class>org.apache.catalina.servlets.CGIServlet</servlet-class>
+<init-param>
+  <param-name>cgiPathPrefix</param-name>
+  <param-value>WEB-INF/cgi</param-value>
+</init-param>
+<init-param>
+  <param-name>executable</param-name>
+  <param-value></param-value>
+</init-param>
+<init-param>
+  <param-name>enableCmdLineArguments</param-name>
+  <param-value>true</param-value>
+</init-param>
+<load-on-startup>5</load-on-startup>
+</servlet>
+```
+
+Also:
+
+```xml
+<servlet-mapping>
+<servlet-name>cgi</servlet-name>
+<url-pattern>/cgi/*</url-pattern>
+</servlet-mapping>
+```
+
+Finally, a script needs to be available in the webapps\ROOT\WEB-INF\cgi directory. For example:
+
+```
+@echo off
+echo Content-Type: text/plain
+echo.
+echo Hello, World!
+```
+
+## Verification Steps
+
+1. Start msfconsole
+2. Do: `use windows/http/tomcat_cgi_cmdlineargs`
+3. Configure rhosts
+4. Configure TARGETURI
+5. Set a payload
+6. Run the module, you should get a session.
+
+## Scenarios
+
+### Check Method
+
+The check method of the exploit explicitly triggers the bug to verify the vulnerable, therefore it should be accurate. To use it, here is an example:
+
+```
+msf5 exploit(windows/http/tomcat_cgi_cmdlineargs) > check
+[+] 172.16.135.141:8080 - The target is vulnerable.
+```
+
+### Code Execution
+
```

Comment:
can you add a h4 with what version of tomcat, what version of java, and what version of windows this was on

Reply:
ah yeah, sure.